apis/r/R/utils-arrow.R

```diff
@@ -178,3 +178,10 @@ arrow_schema_from_tiledb_schema <- function(x) {
   )
   arrow::schema(fields)
 }
+
+#' Validate external pointer to ArrowArray
+#' @noRd
+check_arrow_pointers <- function(arrlst) {
+    stopifnot("First argument must be an external pointer to ArrowArray" = check_arrow_array_tag(arrlst[[1]]),
+              "Second argument must be an external pointer to ArrowSchema" = check_arrow_schema_tag(arrlst[[2]]))
+}
```

Comment:
I am not the hugest fan of moving things from one 'utils' file to another.  I could see consolidating _all_ utils* file into one.   Different personal preferences I suppose.

Reply:
We already have category-specific utils files so I just moved it to the relevant location to make it easier to find.